master/internal/kubernetes/pod.go

```diff
@@ -272,6 +273,42 @@ func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) er
 	return nil
 }
 
+func (p *pod) processMissingPodDeletion(ctx *actor.Context) {
+	ctx.Log().Warn("processing missing pod deletion")
+	if p.container.State == container.Terminated {
+		ctx.Log().Info(
+			"skipping processing missing pod deletion as container is in a terminated state")
+		return
+	}
+
+	// When deleting pods, we do not always receive status updates containing PodFailed
+	// or PodSucceeded. We currently detect this by checking all PodPending messages if
+	// they contain a deletion timestamp.
+	if !p.resourcesDeleted {
+		ctx.Log().Errorf("processing missing pod deletion for a pod that was never deleted")
+	}
+
+	p.container = p.container.Transition(container.Terminated)
+	// Missed pod deletions occur only when a pod is deleted so we assume
```

Comment:
non-blocking: Perhaps something like "when a pod is explicitly deleted by us" or something for emphasis.

Reply:
So this scenario is applicable even if the pod is deleted by the user via `kubectl` or some other mechanism.